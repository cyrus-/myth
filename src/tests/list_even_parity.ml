let s = "
type nat =
  | O
  | S of nat

type bool =
  | True
  | False

type list =
  | Nil
  | Cons of bool * list

let list_even_parity : list -> bool |>
    { [] => True
    | [ False ] => True
    | [ True  ] => False
    | [ False ; False ] => True
    | [ False ; True ] => False
    | [ True  ; False ] => False
    | [ True  ; True ] => True
    } = ?"
