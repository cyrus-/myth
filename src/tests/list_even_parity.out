(* Generated Thu Nov 13 00:35:23 2014 *)
let list_even_parity : list -> bool =
  let rec f1 (l1:list) : bool =
    match l1 with
      | Nil -> True
      | Cons (b1, l2) -> (match f1 l2 with
                            | True -> (match b1 with
                                         | True -> False
                                         | False -> True)
                            | False -> b1)
  in
    f1
;;
